opera/feature_set.go

```diff
@@ -0,0 +1,51 @@
+package opera
+
+import "fmt"
+
+// FeatureSet is an enumeration of different releases, each one enabling a
+// different set of features. These are an abstraction that allows to reason
+// about the different releases instead of isolated upgrades.
+// Feature sets are exclusive, and a (fake-)net cannot be configured with
+// more than one feature set at the time.
+type FeatureSet int
+
+const (
+	SonicFeatures   FeatureSet = iota // < enables the initial sonic release features
+	AllegroFeatures                   // < enables the allegro release features
+)
+
+func (fs FeatureSet) ToUpgrades() (Upgrades, error) {
+	var res Upgrades
+	switch fs {
+	case SonicFeatures:
+		res = Upgrades{
+			Berlin:  true,
+			London:  true,
+			Llr:     false,
+			Sonic:   true,
+			Allegro: false,
+		}
+	case AllegroFeatures:
+		res = Upgrades{
+			Berlin:  true,
+			London:  true,
+			Llr:     false,
+			Sonic:   true,
+			Allegro: true,
+		}
+	default:
+		return res, fmt.Errorf("unknown feature set: %v", fs)
+	}
+	return res, nil
+}
```

Comment:
```suggestion
func (fs FeatureSet) ToUpgrades() (Upgrades, error) {
	res := Upgrades{
				Berlin:  true,
				London:  true,
				Llr:     false,
			}
	switch fs {
	case SonicFeatures:
		res.Sonic := true
	case AllegroFeatures:
		res.Sonic = true
		res.Allegro = true
		}
	default:
		return Upgrade{}, fmt.Errorf("unknown feature set: %v", fs)
	}
	return res, nil
}
```

Considering that for Sonic the "old" 3 will always be the same. Wdyt ?

Reply:
I would rather keep it declarative.  This change obscures the contents. 
If the code grows too much, then we can apply some refactoring.